lib/facility.rb

```diff
@@ -1,14 +1,71 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :facilities, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
+    @facilities = info[:facilities]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
-
-  def add_services(service)
+  
+  def add_service(service)
     @services << service
   end
+
+  def facilities_offering_service(facilities, service)
+    facilities.select { |facility| facility.service.include?(service)}
+  end
+
+  def register_vehicle(vehicle)
```

Comment:
This method is doing a lot! Consider helper methods for 39,40. Promote SRP by breaking this function into a few more smaller functions.